utilities/counted_fs.h

```diff
@@ -0,0 +1,131 @@
+//  Copyright (c) 2016-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <atomic>
+#include <memory>
+
+#include "rocksdb/file_system.h"
+#include "rocksdb/io_status.h"
+#include "rocksdb/rocksdb_namespace.h"
+
+namespace ROCKSDB_NAMESPACE {
+struct FileOpCounters {
+  std::atomic<int> ops;
+  std::atomic<uint64_t> bytes;
+
+  FileOpCounters() : ops(0), bytes(0) {}
+
+  void Reset() {
+    ops = 0;
+    bytes = 0;
+  }
+  void RecordOp(const IOStatus& io_s, size_t added_bytes) {
+    if (!io_s.IsNotSupported()) {
+      ops.fetch_add(1, std::memory_order_relaxed);
+    }
+    if (io_s.ok()) {
+      bytes.fetch_add(added_bytes, std::memory_order_relaxed);
+    }
+  }
+};
+
+struct FileCounters {
```

Comment:
I'm not a big fan of this name, as it sounds like it counts things associated with each file rather than across all files. I like `FileOpCounters` better than `FileCounters` though I don't have great ideas for the other one.

Reply:
Renamed to FileOpCounters